calcom/cal.com
# feat: new webhook 'Form submitted, no event booked'

## What does this PR do?

Adds a new webhook that triggers when no booking was made after routing form was submitted. It will give the user 10 minutes to complete the booking

Webhook payload: 
```
{
  "triggerEvent": "FORM_SUBMITTED_NO_EVENT",
  "createdAt": "2024-10-16T17:04:05.179Z",
  "payload": {
    "formId": "d973e5fc-ef79-4629-b356-818ea232832e",
    "formName": "Test Form",
    "teamId": 2,
    "redirect": {
      "type": "eventTypeRedirectUrl",
      "value": "team/seeded-team/round-robin-seeded-team-event"
    },
    "responses": {
      "Country": {
        "value": "Austria",
        "response": "Austria"
      }
    }
  }
}
```

- Fixes CAL-4496

## How should this be tested?

- create 'form submitted, no event booked' webhook 
- submit a routing from, don't submit a booking 
- see that a new entry is create for tasker that is scheduled for 10 minutes after the form is submitted 
- Change `scheduledAt` of the task to a date in the past 
- run `yarn dev:cron` to run tasker cron job 
- webhook should now trigger
